sdk/containers/azcontainerregistry/blob_client_test.go

```diff
@@ -8,99 +8,221 @@ package azcontainerregistry
 
 import (
 	"bytes"
-	"context"
 	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/streaming"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	"github.com/Azure/azure-sdk-for-go/sdk/internal/mock"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
 	"github.com/stretchr/testify/require"
-	"io"
-	"net/http"
-	"strconv"
-	"strings"
-	"testing"
 )
 
-const alpineBlobDigest = "sha256:042a816809aac8d0f7d7cacac7965782ee2ecac3f21bcf9f24b1de1a7387b769"
-
-func TestBlobClient_CancelUpload(t *testing.T) {
-	startRecording(t)
-	endpoint, cred, options := getEndpointCredAndClientOptions(t)
-	ctx := context.Background()
-	client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+// blobDigest returns the digest of the first blob i.e. layer in the image manifest
+func blobDigest(t *testing.T, image, imageDigest string) string {
+	if recording.GetRecordMode() == recording.PlaybackMode {
+		return fakeDigest
+	}
+	client, err := NewClient("https://"+testConfig.loginServer, testConfig.credential, nil)
 	require.NoError(t, err)
-	startRes, err := client.StartUpload(ctx, "hello-world", nil)
+	res, err := client.GetManifest(ctx, image, imageDigest, &ClientGetManifestOptions{Accept: to.Ptr("application/vnd.docker.distribution.manifest.v2+json")})
 	require.NoError(t, err)
-	_, err = client.CancelUpload(ctx, *startRes.Location, nil)
+	reader, err := NewDigestValidationReader(*res.DockerContentDigest, res.ManifestData)
 	require.NoError(t, err)
+	manifest, err := io.ReadAll(reader)
+	require.NoError(t, err)
+	blobDigest := string(regexp.MustCompile("(sha256:[a-f0-9]{64})").Find(manifest))
+	require.NotEmpty(t, blobDigest)
+	_, hash, found := strings.Cut(blobDigest, ":")
```

Comment:
1.18 feature! :)

Reply:
It was very exciting, truly Cutting edge